code_organization/README.md

```diff
@@ -0,0 +1,229 @@
+# Code Organization
+
+As a file's size grows with continued development, it is important to logically group
+methods, macros, constants, and more to maintain readability. Below are examples of how Red Nova Labs expects files and projects to be organized. Some of these requirements are enforced by pull request [status checks](https://help.github.com/articles/about-required-status-checks/) on GitHub now or will be in the future.
+
+## Ruby on Rails
+
+The structure of a Ruby on Rails project is defined by the framework itself, as outlined [here](http://guides.rubyonrails.org/getting_started.html). Red Nova Labs adheres to this predefined structure with some exceptions. This section of the guide is focused more on how certain types of files within a Rails project should be formatted.
+
+### Controller File Organization
+
+```ruby
+class PersonsController < ApplicationController
+  # Includes
+  include Authenticatable
+
+  # Layout
+  layout 'facility'
+
+  # Macros
+  supports_pjax
+  cache_sweeper :root_sweeper
+  protect_from_forgery with: :exception
+
+  # Actions
+  before_action :authenticate_user
+
+  # GET /
+  def index
+  end
+
+  # POST /
+  def create
+  end
+
+  # GET /new
+  def new
+  end
+
+  # GET /:id/edit
+  def edit
+  end
+
+  # GET /:id
+  def show
+  end
+
+  # PUT /:id
+  # PATCH /:id
+  def update
+  end
+
+  # DELETE /:id
+  def destroy
+  end
+
+  # Protected and Private Methods
+
+  protected
+
+  def some_protected_method
+  end
+
+  private
+
+  def some_private_method
+  end
+end
+```
+
+### Model File Organization
```

Comment:
Where do non-default scopes go in this setup?  I'd suggest they go right after the associations, but I'm not committed to that.

Reply:
I've always put them before public class methods, since they're more or less just chainable class methods.